scripts/gen-api-gtkdoc.py

```diff
@@ -0,0 +1,415 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import re
+from lxml import etree
+from optparse import OptionParser
+
+def normalize_text(s):
+    r"""
+    Normalizes whitespace in text.
+
+    >>> normalize_text("asd xxx")
+    'asd xxx'
+    >>> normalize_text(" asd\nxxx  ")
+    'asd xxx'
+    """
+    return s.replace("\n", " ").strip()
+
+CXX_NAMESPACE_RE = re.compile(r'[_a-zA-Z][_0-9a-zA-Z]*::')
+def fix_definition(s):
+    """
+    Removes C++ name qualifications from some definitions.
+
+    For example:
+
+    >>> fix_definition("bool flag")
+    'bool flag'
+    >>> fix_definition("bool FooBar::flag")
+    'bool flag'
+    >>> fix_definition("void(* _GeanyObjectClass::project_open) (GKeyFile *keyfile)")
+    'void(* project_open) (GKeyFile *keyfile)'
+
+    """
+    return CXX_NAMESPACE_RE.sub(r"", s);
+
+class AtAt(object):
+
+    def __init__(self):
+        self.retval = None
+        self.since = ""
+        self.annot = []
+
+    def cb(type, str):
+        return "@%s %s" % (type, str)
+
+class AtDoc(object):
+    def __init__(self):
+        self.retval = None
+        self.since = ""
+        self.annot = []
+
+    def cb(self, type, str):
+        if (type == "param"):
+            words = str.split(" ", 2);
+            self.annot = []
+        elif (type == "return"):
+            self.annot = []
+        elif (type == "since"):
+            self.since = str.rstrip()
+        elif (type == "geany:skip"):
+            self.annot.append("skip")
+        elif (type == "geany:nullable") or (type == "geany:skip"):
+            self.annot.append(type.split(":")[1])
+        elif (type == "geany:cb"):
+            self.annot.append("scope notified")
+        elif (type == "geany:cbdata"):
+            self.annot.append("closure")
+        elif (type == "geany:cbfree"):
+            self.annot.append("destroy")
```

Comment:
these two are the old version, now should be `geany:closure` and `geany:destroy`.  So, can be merged with the generic `geany:` prefix stripping handling.